### PR TITLE
master Lps 158639 | Add tests on expose erc using assetLibraryId in headless taxonomy

### PR DESCRIPTION
**Pull request by yanan**
**Background**
- Add test to put vocabulary by erc in asset library without erc in body
- Add test to put vocabulary by erc in asset library with erc in body
- Add test to update vocabulary by erc with put request in asset library
- Add test to get vocabulary by erc in asset library
- Add test to get vocabulary by nonexistent erc in asset library
- Add test to delete vocabulary by erc in asset library
- Add test on unable to delete vocabulary by nonexistent erc in asset library

**Test Plan**
- LPS-158840
Given an asset library
When user makes a PUT request to create a vocabulary with assetLibraryId and no ERC specified
Then a vocabulary is created with a generated ERC in response body
- LPS-158841
Given an asset library
When user makes a PUT request to create a vocabulary with assetLibraryId and a custom ERC
Then a vocabulary is created with the custom ERC in response body 
- LPS-158842
Given an asset library
And Given an existing vocabulary
When user makes a PUT request to update the vocabulary name with assetLibraryId and ERC
Then the vocabulary name is updated with the ERC in response body
And Then the previous vocabulary name is not present
- LPS-158843
Given an asset library
And Given an existing vocabulary with ERC
When user makes a GET request to retrieve the vocabulary with assetLibraryId and ERC
Then a vocabulary is retrieved with the ERC in response body
- LPS-158844
Given an asset library
When user makes a GET request to retrieve the vocabulary with assetLibraryId and a nonexisting ERC
Then response body is No AssetVocabulary exists with the key {groupId=${GROUPID}, externalReferenceCode=${ERC}}
- LPS-158845
Given an asset library
And Given an existing vocabulary with ERC
When user makes a DELETE request to delete the vocabulary with ERC
Then the vocabulary is deleted
- LPS-158846 
Given an asset library
And Given an existing vocabulary with ERC
When user makes a DELETE request to delete the vocabulary with a nonexisting ERC
Then no vocabularies are deleted
And Then existing vocabularies are still present

**JIRA Ticket**
- https://issues.liferay.com/browse/LPS-158840 
- https://issues.liferay.com/browse/LPS-158841
- https://issues.liferay.com/browse/LPS-158842
- https://issues.liferay.com/browse/LPS-158843
- https://issues.liferay.com/browse/LPS-158844
- https://issues.liferay.com/browse/LPS-158845
- https://issues.liferay.com/browse/LPS-158846

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:relevant

**Comment by yanan** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro [hunk 1]
```diff
@@ -0,0 +1,136 @@
+definition {
+
+	macro _curlTaxonomyVocabulary {
```
Comment (yanan) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro:3]: All methods with JSONCurlUtil based on this _curlTaxonomyVocabulary() method, in which defined what the api for taxonomy-vocabularies is, as well as a base curl with url, username/password and header etc.

**Comment by yanan** [comment]
Failed test are unrelated to the current pr:
- AdvancedObjectAPI
    - CanReturnAggregationTermsDetailsInOneToManyRelationship
    - CanReturnAggregationTermsDetailsInOneToManyRelationshipAfterObjectDeletion
    - CanReturnAggregationTermsDetailsInOneToManyUpdatedRelationship
- StructuredContentFilteringAndSorting
    - CanFilterGreaterThanPriorityValuesSortedDescending

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:forward

**Pull request closed without merging**